src/Controller/ProductController.php

```diff
@@ -215,4 +215,38 @@ public function add(): string
         header("Location:/");
         return "";
     }
+    /**
+     * Show the products book of the user
+     *
+     * @return string
+     */
+    public function book(): string
+    {
+        if ($this->isConnectedElseRedirection()) {
+            $productManager = new ProductManager();
+            return $this->twig->render("Product/book.html.twig", $productManager->selectForBook($this->user["id"]));
+        }
+
+        return "";
```

Comment:
According to the method isConnectedElseRedirection the book method will never fall into this empty return right ?

Reply:
`isConnectedElseRedirection()` only set the http header `Location` and return if a user is connected because. Grumphp doesn't accept `exit` call after the header is set.